pkg/loki-curator/config/config.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+const ()
+
+// CuratorConfig holds the curator's configurations
+type CuratorConfig struct {
+	LogLevel        string `yaml:"LogLevel,omitempty"`
+	DiskPath        string `yaml:"DiskPath,omitempty"`
+	TriggerInterval int    `yaml:"TriggerInterval,omitempty"`
```

Comment:
What about?
```suggestion
	TriggerInterval time.Duration    `yaml:"TriggerInterval,omitempty"`
```

Reply:
The time.Duration type is int64 and accepts nanoseconds, which I think will be a little bit harder to be configured.
My idea to use Minutes for this configuration because for our use-case it is most relevant (example 60m)

Reply:
It doesn't matter. Duration can be properly parsed.
Seed
```golang
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"path/filepath"
	"time"

	"gopkg.in/yaml.v2"
)

type Config struct {
	Interval time.Duration `yaml:"TriggerInterval,omitempty"`
}

var defaultConfig = Config{
	Interval: time.Second * 60,
}

func parseConfigurations(yamlPath string) (*Config, error) {
	filename, _ := filepath.Abs(yamlPath)
	yamlFile, err := ioutil.ReadFile(filename)
	if err != nil {
		return nil, err
	}

	config := defaultConfig

	if err = yaml.Unmarshal(yamlFile, &config); err != nil {
		return nil, err
	}

	return &config, nil
}

func main() {
	c, err := parseConfigurations(os.Args[1])
	if err != nil {
		panic(err)
	}

	fmt.Printf("Duration: %v, Int64: %d\n", c.Interval, (c.Interval))
}

```
for example. 

You can try it with 
```yaml
TriggerInterval: 5m
```

The output will be
```
Duration: 5m0s, Int64: 300000000000
```

If you decide to use it, I would recommend the one from [k8s](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/duration.go#L27-L29), not from the golang time package.

Reply:
> If you decide to use it, I would recommend the one from [k8s](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/duration.go#L27-L29), not from the golang time package.

Ok, duration from k8s needs some more code and types generation which is overkill for such simple usecase. You can stick to the time.Duration.